mempool/clist_mempool.go

```diff
@@ -662,6 +662,39 @@ func (mem *CListMempool) recheckTxs() {
 	mem.proxyAppConn.FlushAsync()
 }
 
+func (mem *CListMempool) GetNextTxBytes(remainBytes int64, remainGas int64, starter []byte) ([]byte, error) {
+
+	mem.updateMtx.RLock()
+	defer mem.updateMtx.RUnlock()
+
+	var lastElem *clist.CElement
+	var minPriority int64 = 0
+	if len(starter) > 0 {
+		for elem := mem.txs.Front(); elem != nil; elem = elem.Next() {
+			if bytes.Equal(elem.Value.(*mempoolTx).tx.Hash(), starter) {
+				lastElem = elem
+				minPriority = elem.Priority
+				break
+			}
+		}
+	}
+
+	var candidate *clist.CElement
+	for elem := mem.txs.Front(); elem != nil; elem = elem.Next() {
+		mTx := elem.Value.(*mempoolTx)
+		if elem == lastElem || mTx.gasWanted > remainGas || int64(len(mTx.tx)) > remainBytes {
+			continue
+		}
+		if candidate == nil || (elem.Priority >= minPriority && elem.Priority > candidate.Priority) {
+			candidate = elem
+		}
+	}
+	if candidate == nil {
+		return nil, fmt.Errorf("can't find transaction with Bytes %d, Gas %d, starter %s", remainBytes, remainGas, starter)
```

Comment:
usually, errors should be concise, while we can log the detailed failure reasons.

Reply:
e.g. from `core/mempool.go`
`		
env.Logger.Error("Error on broadcastTxCommit", "err", err)
`